### ISSUE TITLE
code-block-tools: indented fenced code blocks (inside admonitions) are not detected

### ISSUE THREAD

**Issue by graelo** [rvben/rumdl#423]
### rumdl version

0.1.18

### Installation method

Homebrew

### Bug description

Hi! First off, thanks for the code-block-tools feature — it's really neat and works great on regular code blocks.

I noticed it doesn't seem to pick up fenced code blocks that are indented inside MkDocs admonitions (or any indented context, really). The external linter/formatter is just not invoked on them, rumdl is invoked instead.

This causes Python comments inside the indented code block to be interpreted as Markdown headings, which causes some unexpected linting warnings, and reformatting mishaps.

I suspect the fence detection doesn't account for the 4-space indent required by admonition content, so the block is never recognized as a fenced code block in this context.

Happy to provide more details or test a fix if that helps!

## Minimal reproduction

Given this Markdown with `flavor = "mkdocs"`:

````markdown
!!! example "A simple function"

    ```python
    # This is a comment, not a heading
    def greet(name: str) -> str:
        # Return a greeting
        return f"Hello, {name}!"
    ```
````

### Expected behavior

`code-block-tools` runs the configured Python linter/formatter (e.g. ruff) on the code block, and the `#` comments are left alone.

### Actual behavior

The code block is ignored by `code-block-tools`. The `# This is a comment` and `# Return a greeting` lines trigger MD023 (Heading should not be indented by x spaces), as if they were Markdown content rather than fenced code.

### Configuration

```toml
flavor = "mkdocs"

[code-block-tools]
enabled = true

[code-block-tools.languages]
python = { lint = ["ruff:check"], format = ["ruff:format"] }
```

### Operating system

macOS

**Comment by Elonmuskfoundation** [comment]
how can i work with you?

**Comment by graelo** [comment]
@Elonmuskfoundation thanks! This `code-block-tools` feature being very new, on my side I first want hear from Ruben (the owner of this repo): may be that's already in the works.

**Comment by rvben** [comment]
@graelo, thanks for the feedback on the `code-block-tools` feature. I will have a look.

**Comment by rvben** [comment]
Fixed in v0.1.20. Fenced code blocks indented inside MkDocs admonitions (!!!), collapsible blocks (???), and tabs (===) are now detected when using the MkDocs flavor. 
Thanks again for raising this, @graelo. 🚀

**Comment by graelo** [comment]
Thank you ❤️ ! It works perfectly

**Comment by rvben** [comment]
Thanks for the feedback, @graelo 💪 Glad to hear
